nexus_constructor/model/transformation.py

```diff
@@ -0,0 +1,72 @@
+import attr
+from PySide2.Qt3DCore import Qt3DCore
+from PySide2.QtGui import QVector3D, QMatrix4x4
+import numpy as np
+
+from nexus_constructor.model.group import Group
+from nexus_constructor.model.helpers import get_item, set_item
+from nexus_constructor.common_attrs import CommonAttrs
+from nexus_constructor.transformation_types import TransformationType
+
+
+class TransformationGroup(Group):
+    def __init__(self):
+        self.nx_class = "NXTransformations"
+
+
+@attr.s
+class Transformation:
+    name = attr.ib(type=str)
+    value = attr.ib()
+    attributes = attr.ib(default={})
+
+    @property
+    def type(self):
+        return get_item(self.attributes, CommonAttrs.TRANSFORMATION_TYPE)
+
+    @type.setter
+    def type(self, new_type):
+        self.attributes[CommonAttrs.TRANSFORMATION_TYPE] = new_type
+
+    @property
+    def vector(self):
+        vector = get_item(self.attributes, CommonAttrs.VECTOR)
+        return (
+            QVector3D(vector[0], vector[1], vector[2]) if vector is not None else None
+        )
+
+    @vector.setter
+    def vector(self, new_vector: QVector3D):
+        vector_as_np_array = np.array([new_vector.x(), new_vector.y(), new_vector.z()])
+        set_item(self.attributes, CommonAttrs.VECTOR, vector_as_np_array)
+
+    @property
+    def ui_value(self):
```

Comment:
Is there a good reason as to why it is called `ui_value`? It is named simply `value` in the NeXus file, right? SO maybe use the same name here.

Reply:
`ui_value` is used for tracking the placeholder value for when the dataset does not have a value castable to float/int, ie a transformation having a stream group as its magnitude

Reply:
I am not convinced by that argument. On the other hand, it is easy to change the name latter so we can leave it as is for now.